lib/tbot/config/config_test.go

```diff
@@ -257,17 +258,18 @@ func TestBotConfig_MarshalYAML(t *testing.T) {
 				AuthServer:      "example.teleport.sh:443",
 				CertificateTTL:  time.Minute,
 				RenewalInterval: time.Second * 30,
-				Outputs: Outputs{
-					&IdentityOutput{
-						Common: OutputCommon{
-							Destination: WrapDestination(&DestinationMemory{}),
-						},
-					},
-				},
 			},
 		},
 	}
+	testYAML(t, tests)
+}
 
+type testYAMLCase[T any] struct {
+	name string
+	in   T
+}
+
+func testYAML[T any](t *testing.T, tests []testYAMLCase[T]) {
```

Comment:
Why so many YAML tests you might ask ? `yaml.v3` has a bunch of weird behaviours (and bugs) that I ran into whilst building out the new polymorphic marshal/unmarshaling. I don't trust myself to not make these mistakes again, and these tests are absurdly easy to regenerate with `GOLDEN_UPDATE=1`, so I consider the overhead of having these tests pretty low.